lib/std/start.zig

```diff
@@ -98,6 +100,22 @@ fn callMain2() noreturn {
     exit2(0);
 }
 
+fn wasmMain2() u8 {
+    switch (@typeInfo(@typeInfo(@TypeOf(root.main)).Fn.return_type.?)) {
+        .Void => {
+            root.main();
+            return 0;
+        },
+        .Int => |info| {
+            if (info.bits != 8 or info.signedness == .signed) {
+                @compileError(bad_main_ret);
```

Comment:
What is `bad_main_ret`? Could this be the source of failures on other targets?

Reply:
https://github.com/ziglang/zig/blob/35f0ab613579efae5801e81d426963b9603db050/lib/std/start.zig#L469

Reply:
Gotcha, thanks!